ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -627,6 +633,23 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
     );
   }
 
+  private Optional<ColumnName> getLeftJoinNonKeyColumn(final Join join) {
+    // This method is called after the verifyJoin() has detected a FK join. This code does not
+    // check for values are in the right order. It is assumed verifyJoin() already checked them
+    // and failed in case they are not valid.
```

Comment:
Looking into `verifyJoin` we also only check if `joinOnNonKeyAttribute(rightExpression, rightNode)` and `joinOnNonKeyAttribute(leftExpression, leftNode)` -- guess we will need to make this more flexible...

Overall, it seems that `getLeftJoinNonKeyColumn` is somewhat similar what adds to my previous comment to push this inside `verifyJoin` (or in particular `joinOnNonKeyAttribute`) ?

Reply:
Just moved it to the verifyJoin.